src/domain/signer.ts

```diff
@@ -0,0 +1,78 @@
+import * as ecc from 'tiny-secp256k1';
+import type { BIP32Interface } from 'bip32';
+import { BIP32Factory } from 'bip32';
+import { Pset, Signer, Transaction, script as bscript, Finalizer, Extractor } from 'liquidjs-lib';
+import { mnemonicToSeed } from 'bip39';
+import type { WalletRepository } from '../infrastructure/repository';
+import { hashPassword, decrypt } from '../utils';
+
+const bip32 = BIP32Factory(ecc);
+const sigValidator = Pset.ECDSASigValidator(ecc);
+
+export class SignerService {
+  private constructor(
+    private walletRepository: WalletRepository,
+    private masterNode: BIP32Interface
+  ) {}
+
+  static async fromPassword(walletRepository: WalletRepository, password: string) {
+    const passwordHash = await walletRepository.getPasswordHash();
+    if (hashPassword(password) !== passwordHash) throw new Error('Invalid password');
+    const encrypted = await walletRepository.getEncryptedMnemonic();
+    if (!encrypted) throw new Error('No mnemonic found in wallet');
+    const decryptedMnemonic = decrypt(encrypted, password);
+    const masterNode = bip32.fromSeed(await mnemonicToSeed(decryptedMnemonic));
+    return new SignerService(walletRepository, masterNode);
+  }
+
+  async signPset(psetBase64: string): Promise<string> {
```

Comment:
Maybe better to take a `Pset` directly instead of `psetBase64`?

Reply:
Internally we can avoid roundtrip de/serialization

Reply:
this idea is also to use this `SignerService` in provider `signPset` method (where pset is passed as base64)

Reply:
You can serialize/deserialize directly in the handler, leaving Pset data structure internally no?